add code of conduct revision

Bradamant3 opened the pull request:
As noted in Trello, this is drawn almost verbatim from the Django CoC. See text inline for further attributions.
This PR is so folks can comment on the approach. It's not ready for merge, because we need at least a revised response/enforcement doc/page to go with it. I'll work on it next, after we agree on the CoC itself.
Once we have the CoC and enforcement updated, we can produce the "Welcome Wagon" content -- some revisions to the current Slack page, plus a Slack welcome message and maybe some other goodies.

Bradamant3 committed:
add code of conduct revision

ericholscher commented:
This looks mostly like a straight port of the Django COC. This is definitely a friendlier and more useful one -- we should probably copy their FAQ's and other tidbits as well -- but I'm +1 on their version, it's much friendlier and to the point.

mjang commented on docs/code-of-conduct.rst:
```diff
@@ -0,0 +1,33 @@
+Code of Conduct
+===============
+
+Like the technical community as a whole, the Write the Docs team and community is made up of a mixture of professionals and volunteers from all over the world, working on every aspect of the mission - including mentorship, teaching and connecting people.
```
I'm not sure about this para -- only the open source tech community includes professionals and volunteers, and "mission" implies that we've defined one... maybe borrow from the main writethedocs.org page:

"Write the Docs is a global community of documentarians who share information, discuss ideas, and work together to improve the art and science of documentation.

yo-bj replied:
I agree on the question about the inclusion of "mission" in regards to community. Maybe we can strike out the rest of the sentence after "world".

On the "professionals and volunteers" phrase, you could count user groups of proprietary software as volunteer efforts. I'm more inclined to let that phrase stay.

mjang commented on docs/code-of-conduct.rst:
```diff
@@ -0,0 +1,33 @@
+
+Like the technical community as a whole, the Write the Docs team and community is made up of a mixture of professionals and volunteers from all over the world, working on every aspect of the mission - including mentorship, teaching and connecting people.
+
+Diversity is one of our huge strengths, but it can also lead to communication issues and unhappiness. To that end, we have a few ground rules that we ask people to adhere to when they're participating within this community and project. These rules apply equally to founders, mentors, admins, and those seeking conversation, discussion, help, and guidance.
```
I'd avoid suggesting that diversity is a weakness... maybe:

"We have ground rules for all members of the community."

yo-bj replied:
Piggy-backing on Mike's comment - maybe we can strike out the first sentence as well as "To that end," and starting the section with "We have a few..."

ericholscher replied:
I think acknowledging that diversity isn't free is useful -- I think "Diversity is one of our huge strengths, but it can also lead to communication issues" is fine -- the unhappiness part reads a bit weird.

kelly-o committed:
Update code-of-conduct.rst

Rewrite on the 'Be considerate' graf + a few minor punctuation and word choice tweaks...

kelly-o commented:
Uh.... I updated with a few changes – I think I did it right? (_github, my nemesis_)

yo-bj commented:
If we were not already planning to do so, forking the [Django reporting procedures](https://github.com/django/code-of-conduct/blob/master/reports.md) would be a good place to start for the reporting piece of the CoC. :)

yo-bj commented on docs/code-of-conduct.rst:
```diff
@@ -0,0 +1,33 @@
+
+* **Be considerate.** Your words and your work will be used by other people, and you in turn will depend on the words and the work of others. Any decision you take will affect users and colleagues, and you should take those consequences into account when making decisions. Remember that we're a world-wide community, so you might not be communicating in someone else's primary language.
+
+* **Be respectful.** Not all of us will agree all the time, but disagreement is no excuse for poor behavior and poor manners. We might all experience some frustration now and then, but we cannot allow that frustration to turn into a personal attack. It’s important to remember that a community where people feel uncomfortable or threatened is not a productive one. Members of the Write the Docs comm ...
```
Dealing with interactions outside the WTD community - does that pertain to those who are tasked with outreach and publicity, or is this meant for any WTD-type interaction between anyone in the community and anyone outside the community?
